history_tool.py

```diff
@@ -0,0 +1,41 @@
+from datetime import datetime
+from collections.abc import AsyncIterable
+
+class History:
+
+    def __init__(self, message :str, date=datetime.now()) -> None:
+        self.message = message
+        self.date = date
+
+    def __lt__(self, other) -> bool:
+        if isinstance(other, self.__class__):
+            return self.date < other.date
+        elif isinstance(other, (list, tuple)):
+            return self.message < other
+        
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self.message == other.message
+        elif isinstance(other, (list, tuple, str)):
+            return self.message == other
+
+    def __repr__(self):
```

Comment:
Why `__repr__`? Convention is that `__repr__` is for programmers and `__str__` is for users, I would rather expect `return f"History(message={self.message}, date={self.date})"` to know exactly what object it is. If You want to log a message by `str()` - which is called when you use f-string formatting, create `__str__`.